packages/backend/media-optimiser/__tests__/media-optimiser.test.ts

```diff
@@ -0,0 +1,131 @@
+    const optimisedMedia = await optimiseMedia({
+      fileName: "png_image",
+      fileBuffer: file,
+      formats: [ "avif", "gif" ],
+      sizes: [ 600 ]
+    });
+
+    for await (const media of optimisedMedia) {
+      const metadata = await sharp(media.optimisedImage).metadata();
+
+      // AVIF images have "heif" format
+      if (media.format === "avif") {
+        expect(metadata.format).toBe("heif");
+        expect(metadata.compression).toBe("av1");
+      } else {
+        expect(metadata.format).toBe(media.format);
+      }
+
+      expect(metadata.width).toBe(media.size);
+    }
+  }, 30000);
+
+  test("Can optimise media with formats and sizes (from heic)", async () => {
+    const file = fs.readFileSync(path.resolve(__dirname, "../assets/heic_image.heic"));
+
+    const optimisedMedia = await optimiseMedia({
+      fileName: "heic_image",
+      fileBuffer: file,
+      formats: [ "avif", "gif" ],
+      sizes: [ 600 ]
+    });
+
+    for await (const media of optimisedMedia) {
+      const metadata = await sharp(media.optimisedImage).metadata();
+
+      // AVIF images have "heif" format
+      if (media.format === "avif") {
+        expect(metadata.format).toBe("heif");
+        expect(metadata.compression).toBe("av1");
+      } else {
+        expect(metadata.format).toBe(media.format);
+      }
+
+      expect(metadata.width).toBe(media.size);
+    }
+  }, 30000);
+
+  test("Can optimise media with formats and sizes (from jpg)", async () => {
+    const file = fs.readFileSync(path.resolve(__dirname, "../assets/jpg_image.jpg"));
+
+    const optimisedMedia = await optimiseMedia({
+      fileName: "jpg_image",
+      fileBuffer: file,
+      formats: [ "avif", "gif" ],
+      sizes: [ 600 ]
+    });
+
+    for await (const media of optimisedMedia) {
+      const metadata = await sharp(media.optimisedImage).metadata();
+
+      // AVIF images have "heif" format
+      if (media.format === "avif") {
+        expect(metadata.format).toBe("heif");
+        expect(metadata.compression).toBe("av1");
+      } else {
+        expect(metadata.format).toBe(media.format);
+      }
+
+      expect(metadata.width).toBe(media.size);
+    }
+  }, 30000);
+
+  test("Can optimise media with formats and sizes (from avif)", async () => {
+    const file = fs.readFileSync(path.resolve(__dirname, "../assets/avif_image.avif"));
+
+    const optimisedMedia = await optimiseMedia({
+      fileName: "avif_image",
+      fileBuffer: file,
+      formats: [ "png", "heif" ],
+      sizes: [ 600 ]
+    });
+
+    for await (const media of optimisedMedia) {
+      const metadata = await sharp(media.optimisedImage).metadata();
+
+      // AVIF images have "heif" format
+      if (media.format === "avif") {
+        expect(metadata.format).toBe("heif");
+        expect(metadata.compression).toBe("av1");
+      } else {
+        expect(metadata.format).toBe(media.format);
+      }
+
+      expect(metadata.width).toBe(media.size);
+    }
+  }, 30000);
+});
```

Comment:
GitHub CI / Lambdas and other dev machines will fail tests if they don't have all of the necessary pre-built binaries.

Perhaps we should ensure that any default formats, and ones that we test, are ones that the default Sharp codec supports: 
`JPEG, PNG, WebP, AVIF, TIFF, GIF (input) and SVG (input) image formats`

But accept other formats in projects like Nokkel where we're expecting `heic, heif` etc.